utils/filesystem.go

```diff
@@ -183,5 +189,69 @@ func UnZip(zipFileName, fileDestination string) {
 			errors.CheckErr(err, 404, "")
 		}
 	}
-	log.Println("File extracted:", zipFileName)
+	fmt.Printf("Extracted file from '%s' to '%s'\n", filePath, destination)
+	return nil
+}
+
+// UnTar unpacks a tar.gz file to a destination
+func UnTar(pathToTarFile, destination string) error {
+	fileReader, err := read(pathToTarFile)
+	if err != nil {
+		return err
+	}
+	defer fileReader.Close()
+	gzipReader, err := gzip.NewReader(fileReader)
+	if err != nil {
+		return err
+	}
+	defer gzipReader.Close()
+	tarReader := tar.NewReader(gzipReader)
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			// end of tar archive
+			break
+		}
+		if err != nil {
+			log.Fatal(err)
+		}
+		target := filepath.Join(destination, header.Name)
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.MkdirAll(target, os.FileMode(header.Mode)); err != nil {
+				log.Fatal(err)
+			}
+		case tar.TypeReg:
+			fileToOverwrite, err := overwrite(target)
+			defer fileToOverwrite.Close()
+			if err != nil {
+				log.Fatal(err)
+			}
+			if _, err := io.Copy(fileToOverwrite, tarReader); err != nil {
+				log.Fatal(err)
+			}
+		default:
+			log.Printf("Can't: %c, %s\n", header.Typeflag, target)
```

Comment:
does this need to go to stderr? Also unsure what this is defaulting too. Maybe a comment?

Reply:
I think it does need to go to stderr because it's a failure (while unpacking a tar.gz). I've clarified the log message